lib/jsdom/level2/html.js

```diff
@@ -2056,7 +2056,28 @@ define('HTMLFrameElement', {
     this._initInsertListener = function () {
       loadFrame(self);
     };
+
+    function onRemove() {
+      self.contentWindow.close();
+
+      // delete frame reference
+      if (self.getAttribute("name")) {
+        parent._frame(self.getAttribute("name"));
+      }
+
+      // delete index reference by moving all frames with a higher index one index down
+      for (var i = self.frameID; i < parent._length - 1; ++i) {
+        parent[i + 1]._frameID--;
+        parent._frame(i, parent[i + 1]);
+      }
+
+      // delete last reference
+      delete parent[parent._length - 1];
+      parent._length--;
+    }
+
     this.addEventListener('DOMNodeInsertedIntoDocument', this._initInsertListener, false);
+    this.addEventListener('DOMNodeRemovedFromDocument', onRemove, false);
```

Comment:
Can we use hooks (if we have one for node removed) instead of mutation events? We already use _atrrModified I noticed.